sensor/common/sensor/central_receiver_test.go

```diff
@@ -0,0 +1,418 @@
+	close(messagesFromCentral)
+
+	s.T().Logf("Waiting for receiever to stop")
+	s.finished.Wait()
+	slowComponent.Stop()
+	_, ok := <-slowComponent.ResponsesC()
+	s.False(ok, "no more message should be processed than what was already read")
+
+	// Verify queue metrics - calculate deltas
+	finalDropped := getOpMetricValue(s.T(), slowComponent.Name(), "Drop")
+	finalAddOperations := getOpMetricValue(s.T(), slowComponent.Name(), "Add")
+	finalRemoveOperations := getOpMetricValue(s.T(), slowComponent.Name(), "Remove")
+
+	droppedDelta := finalDropped - initialDropped
+	addDelta := finalAddOperations - initialAddOperations
+	removeDelta := finalRemoveOperations - initialRemoveOperations
+
+	s.Equal(float64(numberOfCentralMessages-queueSize-1), droppedDelta, "should drop excess messages beyond queue capacity")
+	s.Equal(float64(numberOfCentralMessages-1), addDelta, "should track all add operations")
+	s.Equal(float64(queueSize+1), removeDelta, "should track all remove operations")
+
+	s.NoError(s.receiver.Stopped().Err())
+}
+
+func (s *centralReceiverSuite) Test_ComponentProcessMessageErrorsMetric() {
+	const numberOfCentralMessages = 5
+	errorComponent := &testErrorSensorComponent{testSensorComponent{
+		t:    s.T(),
+		name: "error-component",
+	}}
+
+	components := []common.SensorComponent{errorComponent}
+	s.receiver = NewCentralReceiver(s.finished, components...)
+
+	// Get initial error count before test
+	initialErrors := metrics.GetMetricValue(s.T(), errorsMetric, map[string]string{metrics.ComponentName: "error-component"})
+
+	s.mockClient.EXPECT().Context().AnyTimes().Return(context.Background()).AnyTimes()
+
+	messagesFromCentral := make(chan *central.MsgToSensor)
+	s.mockClient.EXPECT().Recv().MinTimes(numberOfCentralMessages).DoAndReturn(func() (*central.MsgToSensor, error) {
+		msg, ok := <-messagesFromCentral
+		if !ok {
+			s.T().Logf("received EOF from central")
+			return nil, io.EOF
+		}
+		s.T().Logf("received the message from central")
+		return msg, nil
+	})
+
+	s.receiver.Start(s.mockClient)
+	s.T().Logf("Sending %d messages from central", numberOfCentralMessages)
+	for i := 0; i < numberOfCentralMessages; i++ {
+		messagesFromCentral <- testMsg
+	}
+
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		finalErrors := metrics.GetMetricValue(s.T(), errorsMetric, map[string]string{metrics.ComponentName: "error-component"})
+		assert.Equal(c, numberOfCentralMessages, int(finalErrors-initialErrors))
+	}, 5*time.Second, 10*time.Millisecond, "error metric should be incremented when ProcessMessage returns an error")
+
+	s.T().Logf("Sending EOF from central")
+	close(messagesFromCentral)
+
+	s.T().Logf("Waiting for receiever to stop")
+	s.finished.Wait()
+	s.NoError(s.receiver.Stopped().Err())
+}
+
+// testSensorComponent process messages with every tick
+type testSensorComponent struct {
+	t          *testing.T
+	name       string
+	tick       <-chan struct{}
+	responsesC chan *message.ExpiringMessage
+}
```

Comment:
we could change `NewCentralReceiver` to receive a `CentralReceivers` instead of `SensorComponents`. This way we do not need to implement all the functions like `Notify`, `Start`, etc.

```golang
func NewCentralReceiver(finished *sync.WaitGroup, receivers ...common.CentralReceiver) CentralReceiver {
	return &centralReceiverImpl{
		stopper:   concurrency.NewStopper(),
		receivers: receivers,
		finished:  finished,
	}
}
```
I understand that we still need to implement `ResponsesC` and `Stop` for testing purposes.

Reply:
So we can do this but since it's a slice we will need to convert it from []SensorComponenr to []CentraReviver type. I can do it in a separated PR  to not grow this one with a refactor